sqlglot/parser.py

```diff
@@ -1422,6 +1422,18 @@ def _parse_property(self) -> t.Optional[exp.Expression]:
                 this=key,
                 value=self._parse_column() or self._parse_var(any_token=True),
             )
+        else:
```

Comment:
why don't you just match for non strings? why this whole branch?

Reply:
I'll do another pass and try to refactor this

Reply:
Please take another look, tried to simplify things a bit.